ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -0,0 +1,291 @@
+      verifyCompactionDetails(null);
+    }
+
+    minLogSizeToTriggerCompactionInPercentages = new int[]{10, 20, 35, 40, 50, 59};
+    // when used capacity is 60%, compactionDetails is expected to return valid values for all of the above values
+    for (int minLogSize : minLogSizeToTriggerCompactionInPercentages) {
+      properties.setProperty("store.min.log.size.to.trigger.compaction.in.percent", String.valueOf(minLogSize));
+      StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+      MetricRegistry metricRegistry = new MetricRegistry();
+      StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+      blobStore = new MockBlobStore("", config, null, null, metrics, null, 0, null, null, null, time);
+      blobStore.capacityInBytes = CAPACITY_IN_BYTES;
+      blobStore.start();
+      compactionManager = new CompactionManager(config);
+      blobStore.usedCapacity = CAPACITY_IN_BYTES * 6 / 10;
+      blobStore.validLogSegments = generateRandomString(2);
+      verifyCompactionDetails(new CompactionDetails(blobStore.getTime().milliseconds() - messageRetentionTimeInMs,
+          blobStore.validLogSegments));
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} for different values for
+   * {@link StoreConfig#storeDeletedMessageRetentionDays}
+   */
+  @Test
+  public void testDifferentMessageRetentionDays() throws StoreException, InterruptedException {
+    int[] messageRetentionDayValues = new int[]{1, 2, 3, 6, 9};
+    time.sleep(10 * messageRetentionTimeInMs);
+    for (int messageRetentionDays : messageRetentionDayValues) {
+      properties.setProperty("store.deleted.message.retention.days", String.valueOf(messageRetentionDays));
+      StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+      MetricRegistry metricRegistry = new MetricRegistry();
+      StorageManagerMetrics metrics = new StorageManagerMetrics(metricRegistry);
+      blobStore = new MockBlobStore("", config, null, null, metrics, null, 0, null, null, null, time);
+      blobStore.capacityInBytes = CAPACITY_IN_BYTES;
+      compactionManager = new CompactionManager(config);
+      blobStore.usedCapacity = CAPACITY_IN_BYTES * 6 / 10;
+      blobStore.start();
+      blobStore.validLogSegments = generateRandomString(2);
+      verifyCompactionDetails(new CompactionDetails(
+          blobStore.getTime().milliseconds() - messageRetentionDays * Time.SecsPerDay * Time.MsPerSec,
+          blobStore.validLogSegments));
+    }
+  }
+
+  /**
+   * Tests {@link CompactionManager#getCompactionDetails(BlobStore)} failure cases
+   */
+  @Test
+  public void testFailureCases() throws StoreException {
+    // null validLogSegments
+    blobStore.validLogSegments = null;
+    verifyCompactionDetails(null);
+
+    // set valid logSegments
+    blobStore.validLogSegments = generateRandomString(2);
+
+    // blob store not started
```

Comment:
this test belongs in `BlobStore` which should fail if a call to `getLogSegmentsOutsideJournal()` is made. It does not belong here.